extension/persistence/relational-jdbc/src/main/java/org/apache/polaris/extension/persistence/relational/jdbc/JdbcMetaStoreManagerFactory.java

```diff
@@ -106,12 +107,12 @@ private void initializeForRealm(
   }
 
   private DatasourceOperations getDatasourceOperations(boolean isBootstrap) {
-    DatasourceOperations databaseOperations = new DatasourceOperations(dataSource);
+    DatasourceOperations databaseOperations = new DatasourceOperations(dataSource.get());
     if (isBootstrap) {
       // TODO: see if we need to take script from Quarkus or can we just
       // use the script committed in the repo.
       try {
-        databaseOperations.executeScript("scripts/postgres/schema-v1-postgres.sql");
+        databaseOperations.executeScript("postgres/schema-v1-postgresql.sql");
```

Comment:
Ideally the script name should be derived from the DataSource config. Let's inject the DataSource "type" (enum) along with the `DataSource` value and map it to scripts here.

Reply:
I see so essentially type wil indicate the script directory ? and then all the scripts will have same name like for ex schema-v1.sql

```
- h2 
  - schema-v1.sql
- postgres 
    - schema-v1.sql
 ```
 
 so the config we get lets say db type 
 script will be `{type}/schema-v1.sql` ?

Reply:
Follow-up on this : 
1. Do we intent to have custom DS per realm or only 1 DS ? was intially thinking that custom DS name will be name of the realm
2. Type is a bit diff i think i will have to open a connection once per DS and then from the connection get the type and use that to find out which script to use.

Reply:
I'd prefer to autodetect like [this](https://github.com/projectnessie/nessie/blob/de0576ef253df95b03b55f25bf0a50e1b6a3b1f3/versioned/storage/jdbc/src/main/java/org/projectnessie/versioned/storage/jdbc/DatabaseSpecifics.java#L66)

Then each enum value would have a compile-time parameter pointing to the init script.

Reply:
Re 1: Let's go with one DS for this PR and handle multiple DS in a follow-up.

Reply:
> I'd prefer to autodetect like [this](https://github.com/projectnessie/nessie/blob/de0576ef253df95b03b55f25bf0a50e1b6a3b1f3/versioned/storage/jdbc/src/main/java/org/projectnessie/versioned/storage/jdbc/DatabaseSpecifics.java#L66)

yup was referring to the same thing, let me do that.